File: frontend/src/api/agents.ts
import { fetchApi } from './client';

export interface AgentStepResult {
  agent_name: string;
  status: string;
  output: Record<string, any>;
  execution_time_ms: number;
}

export interface AgentWorkflowResponse {
  workflow_id: string;
  buyer_id: string;
  buyer_name: string;
  status: string;
  completed_steps: AgentStepResult[];
  approval_required: boolean;
  summary_action_plan: string;
}

export const executeAgentsApi = (buyerId: string) =>
  fetchApi<AgentWorkflowResponse>('/api/v1/agents/execute', {
    method: 'POST',
    body: JSON.stringify({ buyer_id: buyerId, requires_human_approval: true }),
  });

File: backend/app/schemas/agents.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class AgentExecutionRequest(BaseModel):
    buyer_id: str
    agents: List[str] = ["research", "compliance", "narrative", "outreach", "account_plan"]
    requires_human_approval: bool = True

class AgentStepResult(BaseModel):
    agent_name: str
    status: str
    output: Dict[str, Any]
    execution_time_ms: float

class AgentWorkflowResponse(BaseModel):
    workflow_id: str
    buyer_id: str
    buyer_name: str
    status: str
    completed_steps: List[AgentStepResult]
    approval_required: bool
    summary_action_plan: str
